### PR TITLE
correct quaternion docstring for intrinsic/extrinsic

### LINKED ISSUE #25848: Quaternion.to_euler() mixup of intrinsic vs extrinsic?
The docstring of the function Quaternion.to_euler() (https://github.com/sympy/sympy/blob/d2be7bacd2604e98a642f74028e8f0d7d6084f78/sympy/algebras/quaternion.py#L441-L581) mentions that to result with _intrinsic_ Euler angles, one should provide a sequence of _lowercase_ letters and it should be _uppercase_ for _extrinsic_ rotations.

I believe this should actually be the oposite because:

1. in the linked source code on line 510 of the method `to_Euler()`, the method `_is_extrinsic()` is called. This function contains the statement `extrinsic = seq.islower()` (see line 38 in referenced file). The boolean value of this variable `extrinsic` is returned by the method. The value of the returned value is (at least to my understanding) oposed to the explanation of the docstring.
2. I had already implemented my own code to transform from quaternion to Euler angles and the sympy results match perfectly only if I switch from uppercase to lowercase.

In any case, I'm extremely impressed with all the exelent work that went into this library!

### PR DESCRIPTION
**Pull request by smichr**
#### References to other Issues or PRs

fixes #25848

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by smichr** [comment]
@evbernardes , could you double check this?

File: .mailmap [hunk 1]
```diff
@@ -1224,6 +1224,7 @@ Sagar Bharadwaj <sagarbharadwaj50@gmail.com>
 Sagar231 <sagarfeb298@gmail.com>
 Sahil Shekhawat <sahilshekhawat01@gmail.com>
 Sai Nikhil <tsnlegend@gmail.com>
+Saicharan <62512681+saicharan2804@users.noreply.github.com>
 Saicharan <saicharanhahaha@gmail.com>
```
Comment (smichr) [inline .mailmap:1228]:
```suggestion
Saicharan <saicharanhahaha@gmail.com> <62512681+saicharan2804@users.noreply.github.com>
```

File: .mailmap [hunk 2]
```diff
@@ -1223,8 +1223,7 @@ Sagar Bharadwaj <sagarbharadwaj50@gmail.com>
 Sagar231 <sagarfeb298@gmail.com>
 Sahil Shekhawat <sahilshekhawat01@gmail.com>
 Sai Nikhil <tsnlegend@gmail.com>
-Saicharan <62512681+saicharan2804@users.noreply.github.com>
-Saicharan <saicharanhahaha@gmail.com>
+Saicharan <saicharanhahaha@gmail.com> <62512681+saicharan2804@users.noreply.github.com>
```
Comment (smichr) [inline .mailmap:1226]:
```suggestion
Saicharan <saicharanhahaha@gmail.com>
Saicharan <62512681+saicharan2804@users.noreply.github.com>
```

**Comment by evbernardes** [comment]
> @evbernardes , could you double check this?

Sorry for the delay, but yes, seems to be all good!

**Pull request merged**